tests/jballerina-unit-test/src/test/resources/test-src/expressions/typecast/type_cast_expr.bal

```diff
@@ -265,6 +265,15 @@ function testJsonCastNegative() {
     xml x = xml `text`;
     any a = x;
     assertTypeCastFailureWithMessage(trap <json> a, "incompatible types: 'lang.xml:Text' cannot be cast to 'json'");
+
+    anydata val1 = {"a":1};
+    json|error jval1 = trap <json> val1;
+    test:assertTrue(jval1 is error);
+    if (jval1 is error) {
+        test:assertEquals("{ballerina}TypeCastError", jval1.message());
+        test:assertEquals("incompatible types: 'map<anydata>' cannot be cast to 'json'",
+        <string> checkpanic jval1.detail()["message"]);
+    }
```

Comment:
Shall we add a `record` with the `anydata` field test case similar to the one mentioned in the issue?